gpt_researcher/persistence/repository.py

```
@@ -0,0 +1,352 @@
+        
+        Args:
+            query_embedding: Query embedding vector
+            limit: Maximum number of results
+            similarity_threshold: Minimum similarity score
+            
+        Returns:
+            List of similar documents
+        """
+        # This would require pgvector extension for PostgreSQL
+        # Implementation depends on pgvector being installed
+        async with self.get_session() as session:
```

Ruff (0.12.2)

Local variable `session` is assigned to but never used

Remove assignment to unused variable `session`

(F841)